### ISSUE TITLE
How do I position cursor 

### ISSUE THREAD

**Issue by Toshibass** [HendrikRoth/node-red-contrib-oled#1]
Hi Hendrik, I just installed your oled node and it worked immediately great and potentially very useful,  but I am struggling to understand how to position text, I guess there  must be a way to enter co-ordinates for text positions, for draw line there has be some way to enter start and end position , scroll, font, size etc, etc but currently I have no idea, please advise.

It would be so helpful if there was some information in the node-red info panel detailing the workings of each node and some instructions or examples in your readme file.

Also having a black node with black text is hopeless see attached, it is necessary to be able to identify the name tag of the node .
![capture](https://user-images.githubusercontent.com/6310911/35481231-81d663d0-0417-11e8-8c43-1b0ebccd6a10.JPG)
and finally what is the function of the last node just called 'oled' ? 
![capture1](https://user-images.githubusercontent.com/6310911/35481309-19d450e2-0419-11e8-9733-e2871c00e3c4.JPG)
It may appear that I am criticizing your work .... far from it, you have obviously spent a lot of time producing this node which I am grateful for, its just that I am so thick I cannot figure out how to use it!

Regards Toshi

**Comment by HendrikRoth** [comment]
Hello, just not had the time to write a documentation :-)
You can position the text by passing in an msg.payload object (with a function node for example) like this:

```
msg.payload = {
  x: 50,
  y: 20,
  text: 'Your text'
};

return msg;
```

The `oled` node is a node, that combines the other nodes methods. But i would use in your case the specific nodes you already are using. I hope, this helps.

**Comment by Toshibass** [comment]
aha now I get it thanks for that here my flow up to yet hope it helps others 👍 

[{"id":"d784bbf3.452f88","type":"inject","z":"31d53b23.82f7f4","name":"","topic":"","payload":"Toshi","payloadType":"str","repeat":"","crontab":"","once":false,"x":210,"y":40,"wires":[["ff4ce858.8b6148"]]},{"id":"f248f598.1e9418","type":"inject","z":"31d53b23.82f7f4","name":"","topic":"","payload":"On","payloadType":"str","repeat":"","crontab":"","once":false,"x":210,"y":220,"wires":[["afe8a75.c565758"]]},{"id":"dfb96c5c.5c31f","type":"inject","z":"31d53b23.82f7f4","name":"","topic":"","payload":"Off","payloadType":"str","repeat":"","crontab":"","once":false,"x":210,"y":160,"wires":[["e2e6db12.03ad68"]]},{"id":"937fa93d.8c7ac8","type":"function","z":"31d53b23.82f7f4","name":"time","func":"var d=new Date();\n  var h = d.getHours();\n  var m = d.getMinutes();\n  var s = d.getSeconds();\n  if (h<10){h=\"0\"+h;}\n  if (m<10){m=\"0\"+m;}\n  if (s<10){s=\"0\"+s;}\n  time = (h+\":\"+m+\":\"+s);  \n  \n  msg.payload={size:3,x:1,y:20,text:time};\n  \nreturn msg;","outputs":1,"noerr":0,"x":350,"y":100,"wires":[["423f7137.4b15f"]]},{"id":"757af55.5f6cb0c","type":"inject","z":"31d53b23.82f7f4","name":"","topic":"","payload":"","payloadType":"str","repeat":"1","crontab":"","once":true,"x":210,"y":100,"wires":[["937fa93d.8c7ac8"]]},{"id":"ff4ce858.8b6148","type":"oled-write-string","z":"31d53b23.82f7f4","name":"","display":"5c49c330.36ed7c","x":370,"y":40,"wires":[]},{"id":"afe8a75.c565758","type":"oled-turn-on","z":"31d53b23.82f7f4","name":"","display":"5c49c330.36ed7c","x":380,"y":220,"wires":[]},{"id":"e2e6db12.03ad68","type":"oled-turn-off","z":"31d53b23.82f7f4","name":"","display":"5c49c330.36ed7c","x":380,"y":160,"wires":[]},{"id":"423f7137.4b15f","type":"oled-write-string","z":"31d53b23.82f7f4","name":"","display":"5c49c330.36ed7c","x":510,"y":100,"wires":[]},{"id":"a0b44ce8.ddec9","type":"oled-clear","z":"31d53b23.82f7f4","name":"","display":"5c49c330.36ed7c","x":370,"y":280,"wires":[]},{"id":"34e91d72.81aaa2","type":"inject","z":"31d53b23.82f7f4","name":"clear screen","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"x":190,"y":280,"wires":[["a0b44ce8.ddec9"]]},{"id":"1e65a034.e8ce1","type":"oled-invert","z":"31d53b23.82f7f4","name":"OLED invert","display":"5c49c330.36ed7c","x":370,"y":340,"wires":[]},{"id":"3e339f87.918ce","type":"inject","z":"31d53b23.82f7f4","name":"invert true","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"x":200,"y":340,"wires":[["1e65a034.e8ce1"]]},{"id":"44096a24.8fae94","type":"inject","z":"31d53b23.82f7f4","name":"invert false","topic":"","payload":"false","payloadType":"bool","repeat":"","crontab":"","once":false,"x":200,"y":369,"wires":[["1e65a034.e8ce1"]]},{"id":"8f9c315b.bd849","type":"inject","z":"31d53b23.82f7f4","name":"dim true","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"x":200,"y":420,"wires":[["544b5532.2ebfac"]]},{"id":"55497098.c7b4a","type":"inject","z":"31d53b23.82f7f4","name":"dim false","topic":"","payload":"false","payloadType":"bool","repeat":"","crontab":"","once":false,"x":200,"y":449,"wires":[["544b5532.2ebfac"]]},{"id":"544b5532.2ebfac","type":"oled-dim","z":"31d53b23.82f7f4","name":"OLED dim","display":"5c49c330.36ed7c","x":370,"y":420,"wires":[]},{"id":"5c49c330.36ed7c","type":"oled-config","z":"","name":"","width":"128","height":"64","address":"3c"}]

**Comment by Toshibass** [comment]
But I am stuck on pixel, line and rect nodes ... for example I separately tried each of the following for draw line....
msg.payload={x0:1,y0:20,x1:5,y1:20,color:1};
msg.payload={x0:1,y0:20,x1:5,y1:20};
msg.payload=[{x0:1,y0:20,x1:5,y1:20,color:1}];
msg.payload=[{x0:1,y0:20,x1:5,y1:20}];
msg.payload=[1,20,5,20,1];
msg.payload=[1,20,5,20];
returning msg of each one of those lines results in ReferenceError: node is not defined

please advise thanks.

**Comment by cyrildidier** [comment]
Hi guys.

Toshibass, if you still have not do it by yourself, you can edit the file "oled.js" in ~/..node-red/node_modules/node-red-contrib-oled.

Then in pixel, line and rect functions, you can replace "node" by "self" to solve the issue...

Personally, for example about pixel, I replaced the line "var p = JSON.parse(msg.payload)" by "var p = msg.payload". And I supply the payload by a single point (like: [1,1,1]) or a table of points if I want to draw a picture (like: [[1,1,1],[2,2,1],[3,3,1]])

Enjoy :)

**Comment by cyrildidier** [comment]
Her also the content of the oled.html file to fix node titles issues and switch the node's color to purple:

<script type='text/javascript'>
	RED.nodes.registerType('oled-config', {
		category: 'config',
		defaults: {
			name: {
				value: '',
				require: false
			},
			width: {
				value: 128,
				type: 'number',
				require: true
			},
			height: {
				value: 64,
				type: 'number',
				require: true
			},
			address: {
				value: 0x3D,
				type: 'number',
				require: true
			}
		},
		color:"#d8bfd8",
		label: function() {
			return this.name ? this.name : 'OLED' + this.address
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-config'>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-config-input-name'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-address'><i class='fa'></i> Bus Address</label>
		<input type='text' id='node-config-input-address' />
	</div>
	<div class='form-row'>
		<label for='node-config-input-width'><i class='fa'></i> Width</label>
		<input type='text' id='node-config-input-width' />
	</div>
	<div class='form-row'>
		<label for='node-config-input-height'><i class='fa'></i> Height</label>
		<input type='text' id='node-config-input-height' />
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-write-string', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED String'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-write-string'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-clear', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED Clear'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-clear'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-dim', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED dim'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-dim'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-invert', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED invert'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-invert'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-turn-on', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED Turn on'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-turn-on'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-turn-off', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED Turn off'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-turn-off'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-draw-pixel', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED Draw pixel'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-draw-pixel'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-draw-line', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED Draw line'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-draw-line'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-fill-rect', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED Fill rect'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-fill-rect'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

<script type='text/javascript'>
	RED.nodes.registerType('oled-in', {
		category: 'oled',
		defaults: {
			name: {
				value: '',
				required: false
			},
			display: {
				type: 'oled-config'
			}
		},
		color:"#d8bfd8",
		inputs: 1,
		outputs: 0,
		label: function() {
			return this.name ? this.name : 'OLED in'
		}
	})
</script>

<script type='text/x-red' data-template-name='oled-in'>
	<div class='form-row'>
		<label for='node-config-input-display'><i class='fa'></i> Display</label>
		<input type='text' id='node-input-display'/>
	</div>
	<div class='form-row'>
		<label for='node-config-input-name'><i class='fa fa-tag'></i> Name</label>
		<input type='text' id='node-input-name'/>
	</div>
</script>

**Comment by HendrikRoth** [comment]
Hello, thanks for your work, please make a pull request and i merge the code.

**Comment by Toshibass** [comment]
Hi cyrildidier First of all thank you very much for posting your solution, I managed to modify the oled.html file as per your second post and that works perfectly, however I am having trouble with the modifications to oled.js could you please post a copy of your modified code for this file (like you did for oled.html) or at least the lines you changed, as I am confused which lines need changing from "node" to "self" I tried several combinations including changing all the instance of "node" to "self" but I still get errors.

I appreciate you time spent on this and I would love to get this issue solved, thanks Toshi.

**Comment by cyrildidier** [comment]
Hi Toshibass.

In the node.js please change the two following functions:

	function drawPixel(payload, display, node) {
		if (typeof payload !== 'Array') node.error('Payload not of type Array [ [x, y, color] ]')
		display.drawPixel(payload)
	}

	function OledDrawPixel(n) {
		var self = this
		RED.nodes.createNode(self, n)
		self.display = displays[n.display]
		self.on('input', function(msg) {
			check(self.display, n)
			try {
				var p = JSON.parse(msg.payload)
				drawPixel(p, self.display, node)
			} catch (err) {
				node.error(err)
			}
		})
	}
by
	function drawPixel(payload, display, node) {
		display.drawPixel(payload)
	}

	function OledDrawPixel(n) {
		var self = this
		RED.nodes.createNode(self, n)
		self.display = displays[n.display]
		self.on('input', function(msg) {
			check(self.display, n)
			try {
				var p = msg.payload
				drawPixel(p, self.display, self)
			} catch (err) {
				self.error(err)
			}
		})
	}

Note(1): I removed for testing purpose, in DrawPixel function, the  test to check if inlet data are array (because I now I supply good array :D). so the parameter "node" is not longer usefull but I kept it for futur tests ...
Note(2): I remove the JSON.parse because I provide data by using a table (like payload=[1,1,1] or payload=[[1,1,1],[2,2,1],...] for several pixels at the same time...
Note(3): If you specify color 0 instead of 1 (like [1,1,0]), actually you turn off the pixel's light :). can be usefull if you want to create animation.
Note(4): don't forget to stop and restart node-red to apply modifications ;).

I hope I've help you. I hadn't enough time yet to do lot of tests but the method OledDrawPixel is working perfectly for me like this. I'm goanna work on the other stuff this weekend and I'll give you a feedback :).

**Comment by HendrikRoth** [comment]
Please make a pull request after your last changes, otherwise i can't merge the code clean.

**Comment by cyrildidier** [comment]
Hi Hendrick. I'll do it with pleasure. Right now I haven't test yet all the nodes. That's why I haven't make a pull request. When it will be done I'll give you back what I've changed and you can decide what you want to merge. Untill that I've given some clues to Toshibass to unblock him in his trials ;).

**Comment by cyrildidier** [comment]
Hi Guys,

I've tested all nodes and everything is working. I've added help for all of them and also icons. And also added 2 nodes: one to handle battery animation and another for wifi animation.

If you want to test my modifications **, just be careful to remove nodes from your flow**, if you have, because I simplified the names. 

The files contain "oled.html","oled.js", and the folder icons. After restarting you should get that:

![image](https://user-images.githubusercontent.com/37507102/37866954-eee91a94-2f91-11e8-9743-3e29957e5cb3.png)

I'm going to do a pull request, so, Hendrik, you could merge what you want with your code.

[oled.zip](https://github.com/HendrikRoth/node-red-contrib-oled/files/1844635/oled.zip)

**Comment by cyrildidier** [comment]
and here a flow to test all nodes:

![image](https://user-images.githubusercontent.com/37507102/37866974-5d8b0390-2f92-11e8-97c3-c2fd67f27fb4.png)

[{"id":"4d59e174.4bc57","type":"function","z":"207c6fc3.472cc","name":"Draw 5 pixels","func":"var payload = [[1,1,1],[127,1,1],[127,63,1],[1,63,1],[63,31,1]] ;\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":250,"y":220,"wires":[["15cc42bc.3ce64d"]]},{"id":"fe130cb5.b61fb","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":105,"wires":[["6c8df5b6.5dbebc"]]},{"id":"709cc083.a9d2d","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":70,"wires":[["2a747673.c3581a"]]},{"id":"338be08b.396bd","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":220,"wires":[["4d59e174.4bc57"]]},{"id":"a4ed3ded.eb245","type":"function","z":"207c6fc3.472cc","name":"Just a line","func":"var payload={x0:1,y0:1,x1:127,y1:63,color:1}\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":240,"y":255,"wires":[["a5c6ec65.01b45"]]},{"id":"15cc42bc.3ce64d","type":"Pixel","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":460,"y":220,"wires":[]},{"id":"2a747673.c3581a","type":"Turn-off","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":240,"y":70,"wires":[]},{"id":"6c8df5b6.5dbebc","type":"Invertion","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":240,"y":105,"wires":[]},{"id":"be37f0f1.5e78b","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":35,"wires":[["a1676c58.45e79"]]},{"id":"a1676c58.45e79","type":"Turn-on","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":240,"y":35,"wires":[]},{"id":"c83d09bf.20aec8","type":"Clear","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":230,"y":175,"wires":[]},{"id":"e87fc985.dfa718","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":175,"wires":[["c83d09bf.20aec8"]]},{"id":"68689c8c.826184","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"false","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":140,"wires":[["6c8df5b6.5dbebc"]]},{"id":"a5c6ec65.01b45","type":"Line","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":460,"y":255,"wires":[]},{"id":"4e9a22d.6b567dc","type":"function","z":"207c6fc3.472cc","name":"Say \"coucou\" in double","func":"var payload={size:2,x:1,y:20,text:\"Coucou\"}\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":285,"y":325,"wires":[["14269eab.d24dd1"]]},{"id":"14269eab.d24dd1","type":"String","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":460,"y":325,"wires":[]},{"id":"9268fb3b.79fc48","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":255,"wires":[["a4ed3ded.eb245"]]},{"id":"752ae36.f7b511c","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":325,"wires":[["4e9a22d.6b567dc"]]},{"id":"9b8239fc.2a81c8","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":375,"wires":[["7016a169.d391e"]]},{"id":"7e514651.3aa3a8","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"false","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":410,"wires":[["7016a169.d391e"]]},{"id":"7016a169.d391e","type":"Dimmed","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":245,"y":375,"wires":[]},{"id":"a11b41df.2ffd9","type":"Battery","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":380,"y":460,"wires":[]},{"id":"2113440.67be2bc","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":460,"wires":[["48b3a8dc.261ab8"]]},{"id":"48b3a8dc.261ab8","type":"function","z":"207c6fc3.472cc","name":"x,y,100%","func":"var payload={x:105,y:2,p:100};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":235,"y":460,"wires":[["a11b41df.2ffd9"]]},{"id":"87f5dde5.0cea9","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":495,"wires":[["243b2c3d.8ddae4"]]},{"id":"243b2c3d.8ddae4","type":"function","z":"207c6fc3.472cc","name":"x,y,50%","func":"var payload={x:105,y:2,p:50};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":235,"y":495,"wires":[["a11b41df.2ffd9"]]},{"id":"21d9904b.536cb","type":"Wifi","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":370,"y":580,"wires":[]},{"id":"514745fb.d17d1c","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":580,"wires":[["553dec70.ba7794"]]},{"id":"553dec70.ba7794","type":"function","z":"207c6fc3.472cc","name":"x,y,100%","func":"var payload={x:1,y:1,p:100};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":235,"y":580,"wires":[["21d9904b.536cb"]]},{"id":"98b256e2.ddab18","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":650,"wires":[["5860b633.c87588"]]},{"id":"5860b633.c87588","type":"function","z":"207c6fc3.472cc","name":"x,y,5%","func":"var payload={x:1,y:1,p:5};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":225,"y":650,"wires":[["21d9904b.536cb"]]},{"id":"87947552.5eeea8","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":530,"wires":[["52802338.45a47c"]]},{"id":"52802338.45a47c","type":"function","z":"207c6fc3.472cc","name":"x,y,5%","func":"var payload={x:105,y:2,p:5};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":225,"y":530,"wires":[["a11b41df.2ffd9"]]},{"id":"9b684c1c.4f1b9","type":"function","z":"207c6fc3.472cc","name":"x,y,50%","func":"var payload={x:1,y:1,p:50};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":235,"y":615,"wires":[["21d9904b.536cb"]]},{"id":"3afb92b.1b5846e","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":615,"wires":[["9b684c1c.4f1b9"]]},{"id":"c70bddb4.10cf","type":"function","z":"207c6fc3.472cc","name":"Just a filled rectangle","func":"var payload={x:30,y:15,w:64,h:32,color:1}\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":280,"y":290,"wires":[["38327f80.5d3de"]]},{"id":"ad2c2886.c05bf8","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":290,"wires":[["c70bddb4.10cf"]]},{"id":"38327f80.5d3de","type":"FillRectangle","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":480,"y":290,"wires":[]},{"id":"6b1f520c.c68efc","type":"Scroll","z":"207c6fc3.472cc","name":"","display":"cfcfc006.f7e52","x":525,"y":700,"wires":[]},{"id":"c4d5995f.6b2ea8","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":735,"wires":[["1332367b.c42f5a"]]},{"id":"1332367b.c42f5a","type":"function","z":"207c6fc3.472cc","name":"Scroll left for rows (start,stop)","func":"var payload={direction:\"left\",start:10,stop:63};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":295,"y":735,"wires":[["6b1f520c.c68efc"]]},{"id":"c8f94020.4bfea","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"false","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":700,"wires":[["6b1f520c.c68efc"]]},{"id":"f0be0cc0.521c8","type":"function","z":"207c6fc3.472cc","name":"Scroll righ for rows (start,stop)","func":"var payload={direction:\"right\",start:10,stop:63};\nvar msg = {payload:payload};\nreturn msg;","outputs":1,"noerr":0,"x":305,"y":770,"wires":[["6b1f520c.c68efc"]]},{"id":"43b4e540.6e81cc","type":"inject","z":"207c6fc3.472cc","name":"","topic":"","payload":"true","payloadType":"bool","repeat":"","crontab":"","once":false,"onceDelay":0.1,"x":105,"y":770,"wires":[["f0be0cc0.521c8"]]},{"id":"cfcfc006.f7e52","type":"oled-config","z":"","name":"oled","width":"128","height":"64","address":"3c"}]

**Comment by cyrildidier** [comment]
**And, at last, Hendrick, don't hesitate to correct my poor English in the help sections :)**

**Comment by HendrikRoth** [comment]
Thanks for your work. Committed and pushed.

**Comment by Toshibass** [comment]
Hi cyrildidier 

You made a S U P E R B  job .... I managed to get everything working despite being unable to import the above flow something is wrong with the code but I managed to dig out bits of it to recreate all of the flow.

This enhancement has made this node very very useful thanks for all of your work and time.

HendrikRoth although you say  Committed and pushed. "version": "0.1.0"  when I try to install it I still get "0.0.2"  perhaps theirs a delay and it takes some time to appear .

Anyway thanks for everything.... best regards Toshi Bass
